File: src/org/stormgears/utils/dsio/ITernarySwitch.java
package org.stormgears.utils.dsio;

public interface ITernarySwitch extends ISwitch {
	void whenFlippedTernary(TernaryFlipListener listener);

	void whenFlipped(TernaryFlipListener listener);

	enum SwitchState {
		Up,
		Neutral,
		Down
	}

	@FunctionalInterface
	interface TernaryFlipListener {
		void flipped(SwitchState state);
	}
}

File: src/main/java/org/stormgears/utils/dsio/ITernarySwitch.kt
package org.stormgears.utils.dsio

interface ITernarySwitch : ISwitch {
	fun whenFlippedTernary(listener: TernaryFlipListener)
	fun whenFlippedTernary(listener: (state: SwitchState) -> Unit)

	fun whenFlipped(listener: TernaryFlipListener)

	enum class SwitchState {
		Up,
		Neutral,
		Down
	}

	@FunctionalInterface
	interface TernaryFlipListener {
		fun flipped(state: SwitchState)
	}
}
